tests/test_otdd.py

```diff
@@ -0,0 +1,109 @@
+def metrics_config() -> dict:
+    project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+    metrics_config_path = os.path.join(
+        project_root, "tests", "testconfig", "metrics.yaml"
+    )
+    otdd_config = load_configs(metrics_config_path)["metrics_config"]
+    # filter down to only otdd configs
+    otdd_config = {k: v for k, v in otdd_config.items() if v["class"] == "otdd"}
+
+    return otdd_config
+
+
+# This test checks that the distance between a dataset and itself is close to the
+# expected value within a tolerance
+# A check for equality is not performed as different processors can return slightly
+# different values
+# Approximate methods may be non-zero when comparing a dataset to itself and these
+# are checked against the known value for the seed
+@pytest.mark.skipif(
+    platform.processor() == "arm",
+    reason="These tests should not be run on Apple M1 devices",
+)
+def test_cifar_otdd_same(metrics_config: dict):
+    dmpair = DMPair(metrics_config=metrics_config, seed=42)
+    similarity_dict = dmpair.compute_similarity(only_train=False)
+    similarity_dict_only_train = dmpair.compute_similarity(only_train=True)
+    test_scenarios = {
+        "same_result": similarity_dict,
+        "same_result_only_train": similarity_dict_only_train,
+    }
+    failures = []
+    for scenario, results in test_scenarios.items():
+        for k in metrics_config:
+            test_name = k + "/" + scenario
+            expected_result = metrics_config[k]["expected_results"][scenario]
+            actual_result = results[k]
+            if not np.isclose(actual_result, expected_result, rtol=1e-5, atol=1e-8):
+                failure = (
+                    "test:"
+                    + test_name
+                    + ", expected result: "
+                    + str(expected_result)
+                    + ", actual result: "
+                    + str(actual_result)
+                )
+                failures.append(failure)
+    if failures:
+        pytest.fail("\n".join(failures))
+
+
+# This test checks that the distance between two different datasets is close to the
+# expected value within a tolerance
+# A check for equality is not performed as different processors can return slightly
+# different values
+# The calculated distance is checked against the known value for the seed - brittle
+# test but useful for messing with code
+@pytest.mark.skipif(
+    platform.processor() == "arm",
+    reason="These tests should not be run on Apple M1 devices",
+)
+def test_cifar_otdd_different(metrics_config: dict):
+    dmpair = DMPair(metrics_config=metrics_config, drop_percent_A=0.2, seed=42)
+    similarity_dict = dmpair.compute_similarity(only_train=False)
+    similarity_dict_only_train = dmpair.compute_similarity(only_train=True)
+    test_scenarios = {
+        "diff_result": similarity_dict,
+        "diff_result_only_train": similarity_dict_only_train,
+    }
+    failures = []
```

Comment:
I like the reporting that we get out of this approach.
You might want to consider using something like `pytest-check` that achieves a similar outcome. 
```suggestion
    from pytest_check import check # move this up
    
    for scenario, results in test_scenarios.items():
        for k in metrics_config:
            expected_result = metrics_config[k]["expected_results"][scenario]
            actual_result = results[k]

            with check:
                assert np.isclose(actual_result, expected_result, rtol=1e-5, atol=1e-8)
```

if you run pytest in verbose then you still get the detailed output of mismatched values as you've captured here.

If y

Reply:
happy to keep this if you'd prefer not to introduce another dependency.
Whichever approach is taken, we should do the same in the mmd tests.